langchain/schema/output_parser.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Generic, TypeVar
+
+from pydantic import BaseModel
+
+from langchain.schema.prompts import PromptValue
+
+T = TypeVar("T")
+
+
+class BaseOutputParser(BaseModel, ABC, Generic[T]):
```

Comment:
We should check whether it would make sense to add an on-ramp from a `Callable[[str], T]`.